### PR TITLE
Add examples of using Flink with different catalogs

### PR DESCRIPTION
**Pull request by rmoff**
- **Add examples of using Flink with different catalogs**

### CODE REVIEW

File: catalogs/flink-iceberg-hive/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    build: ./hms-standalone-s3
+    ports:
+      - "9083:9083"
+    environment:
+      - HMS_LOGLEVEL=INFO
+
+# Without a network explicitly defined, you hit this Hive/Thrift error
+# java.net.URISyntaxException Illegal character in hostname
+# https://github.com/TrivadisPF/platys-modern-data-platform/issues/231
+networks:
```
Comment (gunnarmorling) [inline catalogs/flink-iceberg-hive/docker-compose.yml:48]: Shouldn't the services above in this network then?

File: catalogs/flink-iceberg-hive/flink/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+RUN apt-get update && \
+    apt-get install -y vim lnav
+
+RUN echo "Add Flink S3 Plugin" && \
+    mkdir ./plugins/s3-fs-hadoop && \
+    cp ./opt/flink-s3-fs-hadoop-1.18.1.jar ./plugins/s3-fs-hadoop/
+
+RUN echo "-> Install JARs: Flink's Hive connector" && \
+    mkdir -p ./lib/hive && \
+    curl https://repo1.maven.org/maven2/org/apache/flink/flink-sql-connector-hive-3.1.3_2.12/1.18.1/flink-sql-connector-hive-3.1.3_2.12-1.18.1.jar -o ./lib/hive/flink-sql-connector-hive-3.1.3_2.12-1.18.1.jar
```
Comment (gunnarmorling) [inline catalogs/flink-iceberg-hive/flink/Dockerfile:23]: Here I'd use the `-O` option I mentioned in the blog post the other day, to keep things concise (and easier to update later on).

**Review by gunnarmorling — APPROVED** [review]
Nice one! I haven't run it, but it looks great. Some minor comments inline. And can you add these examples to the list in the root README? Cool stuff!

File: catalogs/flink-iceberg-hive/flink/Dockerfile (hunk 2 above)
Comment (rmoff) [reply catalogs/flink-iceberg-hive/flink/Dockerfile:23]: Done. thanks.

File: catalogs/flink-iceberg-hive/docker-compose.yml (hunk 1 above)
Comment (rmoff) [reply catalogs/flink-iceberg-hive/docker-compose.yml:48]: I think it happens implicitly.

**Pull request merged**